frame/support/procedural/src/construct_runtime_args.rs

```diff
@@ -0,0 +1,60 @@
+mod keyword {
+	syn::custom_keyword!(unique_id);
+}
+
+pub fn decl_construct_runtime_args(input: proc_macro::TokenStream) -> proc_macro::TokenStream {
+	use frame_support_procedural_tools::generate_crate_access;
+
+	let def = syn::parse_macro_input!(input as DeclArgsDef);
+	let args = &def.args;
+
+	// frame-support is made available by construct_runtime_preprocess
+	let hidden_crate_name = "construct_runtime_preprocess";
+	let scrate = generate_crate_access(&hidden_crate_name, "frame-support");
+
+	let hidden_macro_name = syn::Ident::new(
+		&format!("__hidden_unique_construct_runtime_args_{}", def.unique_id),
+		proc_macro2::Span::call_site(),
+	);
+
+	quote::quote!(
+		/// This can be internally called by `construct_runtime` to builds the pallet args.
+		#[macro_export]
+		#[doc(hidden)]
+		macro_rules! #hidden_macro_name {
```

Comment:
can you show point out where this is used? I couldn't find it anywhere.

Reply:
it is used in the preprocess of `construct_runtime!`
```rust
			expand = quote_spanned!(module_name.span() =>
				#module_module::construct_runtime_args! {
					{ #module_name : #module_module #module_instance}
					#expand
				}
			);
```